config/config.go

```diff
@@ -508,6 +522,49 @@ func (c *ServersetSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
 	return checkOverflow(c.XXX, "serverset_sd_config")
 }
 
+// KubernetesSDConfig is the configuration for Kubernetes service discovery.
+type KubernetesSDConfig struct {
+	Server           string   `yaml:"server"`
+	NodeMetricsPort  int      `yaml:"node_metrics_port,omitempty"`
+	InCluster        bool     `yaml:"in_cluster,omitempty"`
+	BearerTokenFile  string   `yaml:"bearer_token_file,omitempty"`
+	Username         string   `yaml:"username,omitempty"`
+	Password         string   `yaml:"password,omitempty"`
+	Insecure         bool     `yaml:"insecure,omitempty"`
+	CertFile         string   `yaml:"cert_file,omitempty"`
+	KeyFile          string   `yaml:"key_file,omitempty"`
+	CAFile           string   `yaml:"ca_file,omitempty"`
+	RetryDelay       Duration `yaml:"retry_delay,omitempty"`
+	RequestTimeout   Duration `yaml:"request_timeout,omitempty"`
+	DiscoverServices bool     `yaml:"discover_services,omitempty"`
```

Comment:
We thought about allowing a way to grab all services from Consul too, just couldn't decide on how to configure it 100%. A boolean probably works fine (again, consistency is key - so this would also be a decision for Consul).
Consul however allows for a list of explicit services too.

Technically this is another case that can be solved with relabeling, but is common enough to allow direct configuration.
Do you think it would make sense in the context of Kubernetes too?